### PR TITLE
[RLlib] Tune trial + checkpoint selection example.

### PR DESCRIPTION
**Pull request by sven1977**
This PR adds an example script on how to select Tune trials and checkpoints by different custom criteria.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/tune/analysis/experiment_analysis.py [hunk 1]
```diff
@@ -238,7 +238,16 @@ def get_trial_checkpoints_paths(self,
...
+            ret = []
+            for c in checkpoints:
+                # Support metrics given as paths, e.g.
+                # "info/learner/default_policy/policy_loss".
+                path = metric.split("/")
+                result = c.result
+                for p in path:
+                    result = result[p]
+                ret.append((c.value, result))
+            return ret
```
Comment (krfricke) [inline python/ray/tune/analysis/experiment_analysis.py:250]: Can we use

```
from ray.tune.utils.util import unflattened_lookup

return [(c.value, unflattened_lookup(metric, c.result)) for c in checkpoints]
```

here?

Comment (sven1977) [reply python/ray/tune/analysis/experiment_analysis.py:250]: Ah, cool. No idea this existed :)

Comment (sven1977) [reply python/ray/tune/analysis/experiment_analysis.py:250]: Done.

**Review by krfricke — COMMENTED** [review]
Looks great, but the test is failing due to resource constraints:

```
ray.tune.error.TuneError: Insufficient cluster resources to launch trial: trial requested 3 CPUs, 0 GPUs, but the cluster has only 2 CPUs, 0 GPUs, 3.71 GiB heap, 1.27 GiB objects (1.0 node:10.30.1.54). 
```

**Comment by sven1977** [comment]
@krfricke, thanks, this should be fixed now.

**Review by krfricke — APPROVED** [review]

**Pull request merged**